libraries/3rdparty/lwip/src/portable/arch/sys_arch.c

```diff
@@ -129,11 +129,9 @@ void sys_mbox_free( volatile sys_mbox_t *pxMailBox )
         pxMailBox->xMbox = NULL;
         taskEXIT_CRITICAL();
 
-        while( xTask != NULL )
+        if( xTask != NULL )
         {
             xTaskAbortDelay( xTask );
```

Comment:
I'm trying to understand how this code is suppose to work. What happens if xTaskAbortDelay() unblocks  xTask, but pxMailBox->xTask is not set to NULL; i.e.;  xTask gets blocked again on a receive message.

Reply:
In the fetching task, the next step after the blocking operation is to reread the mbox pointer.  If the fetching task was unblocked by this xTaskAbortDelay(), then the mbox pointer has been set to NULL, which prevents another read on the mbox.